src/services/matchmaking/matchmaking_service.cpp

```diff
@@ -113,4 +138,162 @@ namespace big
+					rage::rlTaskStatus status;
+					MatchmakingId multiplexed_id;
+					if (!g_hooking->get_original<hooks::advertise_session>()(0, num_slots, available_slots, attributes, -1, info, &multiplexed_id, &status))
+					{
+						LOG(WARNING) << __FUNCTION__ ": advertise_session returned false for multiplex task " << i;
+						return;
+					}
+					
+					while (status.status == 1)
+						script::get_current()->yield();
+
+					if (status.status == 2)
+					{
+						LOG(WARNING) << __FUNCTION__ ": advertise_session failed for multiplex task " << i;
+						return;
+					}
+
+					if (auto it = m_multiplexed_sessions.find(id_hash); it != m_multiplexed_sessions.end())
+					{
+						it->second.push_back(multiplexed_id);
+					}
+					else
+					{
+						LOG(WARNING) << __FUNCTION__ ": created a multiplexed session advertisement, but the primary advertisement no longer exists";
+					}
+				});
+			}
+			status->status = 3; // return success for original caller
+		});
+
+		return true;
+	}
+
+	void matchmaking_service::handle_update(int num_slots, int available_slots, rage::rlSessionInfo* info, MatchmakingAttributes* attributes, MatchmakingId* id)
+	{
+		patch_matchmaking_attributes(attributes);
+
+		// this can be fire and forget, but it's probably a good idea to be notified if something goes wrong
+		if (auto it = m_multiplexed_sessions.find(rage::joaat(id->m_data1)); it != m_multiplexed_sessions.end())
+		{
+			if (!g.spoofing.multiplex_session)
+			{
+				// option disabled mid-session
+				return;
+			}
+
+			int i = 0;
+			for (auto& multiplex_session : it->second)
+			{
+				g_fiber_pool->queue_job([&multiplex_session, num_slots, available_slots, info, attributes, i] {
+					rage::rlTaskStatus status;
+					if (!g_hooking->get_original<hooks::update_session_advertisement>()(0, &multiplex_session, num_slots, available_slots, info, attributes, &status))
+					{
+						LOG(WARNING) << __FUNCTION__ ": update_session_advertisement returned false for multiplex task " << i;
+						return;
+					}
+
+					while (status.status == 1)
+						script::get_current()->yield();
+
+					if (status.status == 2)
+					{
+						LOG(WARNING) << __FUNCTION__ ": update_session_advertisement failed for multiplex task " << i;
+						return;
+					}
+
+				});
+				i++;
+			}
+		}
+	}
+
+	bool matchmaking_service::handle_unadvertise(MatchmakingId* id)
+	{
+		if (auto it = m_multiplexed_sessions.find(rage::joaat(id->m_data1)); it != m_multiplexed_sessions.end())
+		{
+			for (auto& multiplex_session : it->second)
+			{
+				g_hooking->get_original<hooks::unadvertise_session>()(0, &multiplex_session, nullptr);
+			}
+
+			m_multiplexed_sessions.erase(it);
+			return false;
+		}
+		else
+		{
+			if (g.spoofing.multiplex_session)
+				for (auto& [_, children] : m_multiplexed_sessions)
+					for (auto& session : children)
+						if (session.m_data2 == id->m_data2 && session.m_data3 == id->m_data3)
+							return true; // prevent auto cleanup

```

Comment:
I don't like the way this looks, I think you should add curley braces. Anything more than 3 nested statements becomes too much.